src/mscorlib/src/System/ReadOnlySpan.cs

```diff
@@ -328,7 +328,7 @@ public T[] ToArray()
         public static ReadOnlySpan<T> Empty => default(ReadOnlySpan<T>);
     }
 
-    public static class ReadOnlySpanExtensions
+    public static class ReadOnlySpanExtensionMethods
```

Comment:
There are two kinds of extension methods:

- Low-level extension methods that need access to runtime internals. Moving these to corefx is a hassle because of them you need to expose the runtime internals instead.
- A large number of high-level extension methods. We want to have them in corefx to allow duplicated sources.

I think we want to have several different types for different kinds of extension methods, as it is being discussed in the email thread parallel to this issue. Krzysztof and Immo should make a call on the naming.